lib/nerve.rb

```diff
@@ -86,15 +165,47 @@ def run
       $EXIT = true
     end
 
-    def launch_watcher(name, config)
-      log.debug "nerve: launching service watcher #{name}"
-      watcher = ServiceWatcher.new(config.merge({'instance_id' => @instance_id, 'name' => name}))
-      @watchers[name] = Thread.new{watcher.run}
+    def merged_config(config, name)
+      # Get a deep copy so sub-hashes are properly handled
+      deep_copy = Marshal.load(Marshal.dump(config))
+      return deep_copy.merge({'instance_id' => @instance_id, 'name' => name})
+    end
+
+    def launch_watcher(name, config, wait=false)
```

Comment:
i would turn this final argument into an opt hash:

```ruby
def launch_watcher(name, config, opts = {})
  wait = opts[:wait] || false
end
```

then invocations are more clear; instead of `launch_watcher(name, config, true)` it would be `launch_watcher(name, config, :wait => true)`